File: dehaze-python/app/models/entity/sys_ai_skill.py
"""AI 对话 Skill 主表实体（F-M08-006 Skills 管理）。

对应表 sys_ai_skill：承载 Skill 元数据与 SKILL.md 指令正文（遵循业界 Agent Skills 规范）。
skill_name 与 sys_ai_agent_skill.skill_name 保持外键语义（不加物理外键，对齐项目惯例）；
status 标识启停（0=禁用，1=启用，对齐 SDK SkillVO.status），禁用后不出现在 SkillManager 索引中。
license/compatibility/metadata/allowed_tools 对应 SKILL.md frontmatter 可选字段。
SKILL 目录内的其余文件（reference/ script/ assets/ README.md）存 sys_ai_skill_file 子表。
配置类表使用逻辑删除（SoftDeleteMixin），删除前须校验被 Agent 关联。
"""

from sqlalchemy import BigInteger, Index, JSON, SmallInteger, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import BaseModel, SoftDeleteMixin


class SysAiSkill(BaseModel, SoftDeleteMixin):
    __tablename__ = "sys_ai_skill"
    __table_args__ = (
        Index("uk_name", "name", unique=True),
        Index("idx_status", "status"),
        {"comment": "AI对话Skill主表(F-M08-006)"},
    )

    id: Mapped[int] = mapped_column(
        BigInteger, primary_key=True, autoincrement=True, comment="主键"
    )
    name: Mapped[str] = mapped_column(
        String(128), nullable=False, comment="Skill名称(唯一,关联sys_ai_agent_skill.skill_name)"
    )
    description: Mapped[str] = mapped_column(
        String(500), nullable=False, comment="Skill描述(供LLM渐进式加载索引使用)"
    )
    scene: Mapped[str] = mapped_column(
        String(255), nullable=False, default="", comment="适用场景(前端筛选/展示)"
    )
    instruction: Mapped[str | None] = mapped_column(
        Text, nullable=True, comment="SKILL.md指令正文(skill_load时注入,frontmatter之外的内容)"
    )
    license: Mapped[str | None] = mapped_column(
        String(255), nullable=True, comment="SKILL.md frontmatter license(许可证)"
    )
    compatibility: Mapped[str | None] = mapped_column(
        String(500), nullable=True, comment="SKILL.md frontmatter compatibility(环境要求)"
    )
    # 列名 metadata（frontmatter 键），Python 属性名 skill_metadata 避开 Declarative 保留字
    skill_metadata: Mapped[dict | None] = mapped_column(
        "metadata", JSON, nullable=True, comment="SKILL.md frontmatter metadata(任意键值,如版本/作者)"
    )
    allowed_tools: Mapped[str | None] = mapped_column(
        String(500),
        nullable=True,
        comment="SKILL.md frontmatter allowed-tools(预批准工具列表,空格分隔)",
    )
    status: Mapped[int] = mapped_column(
        SmallInteger, nullable=False, default=1, comment="启停状态(0:禁用;1:启用)"
    )
    source: Mapped[str] = mapped_column(
        String(32),
        nullable=False,
        default="admin",
        comment="来源(builtin:内置播种;admin:管理员创建)",
    )
    market_shared: Mapped[int] = mapped_column(
        SmallInteger,
        nullable=False,
        default=0,
        comment="是否共享至Skill市场(0:否;1:是)",
    )

File: dehaze-go/internal/model/sys_ai_skill.go
package model

import "encoding/json"

// SysAiSkill Skill 主表（F-M08-006 Skills 管理）
type SysAiSkill struct {
	BaseModel
	Name          string          `gorm:"column:name;type:varchar(128);not null;comment:Skill名称(唯一)" json:"name"`
	Description   string          `gorm:"column:description;type:varchar(500);not null;comment:Skill描述" json:"description"`
	Scene         string          `gorm:"column:scene;type:varchar(255);not null;default:'';comment:适用场景" json:"scene"`
	Instruction   *string         `gorm:"column:instruction;type:text;comment:SKILL.md指令正文" json:"instruction"`
	License       *string         `gorm:"column:license;type:varchar(255);comment:frontmatter license" json:"license"`
	Compatibility *string         `gorm:"column:compatibility;type:varchar(500);comment:frontmatter compatibility" json:"compatibility"`
	Metadata      json.RawMessage `gorm:"column:metadata;type:json;comment:frontmatter metadata" json:"metadata"`
	AllowedTools  *string         `gorm:"column:allowed_tools;type:varchar(500);comment:frontmatter allowed-tools" json:"allowedTools"`
	// 不带 `default:` 标签：GORM 对带默认值的字段在零值时会**用默认值替换**（并省略列），
	// 导致"新建 Skill 默认禁用"（status=0，python `_STATUS_DISABLED`）永远插不进去 0。
	// 列本身的 DEFAULT 1 仍在（不显式给值时才生效），语义与 python 显式赋 0 一致。
	Status       int8   `gorm:"column:status;type:tinyint;not null;comment:启停状态(0:禁用;1:启用)" json:"status"`
	Source       string `gorm:"column:source;type:varchar(32);not null;default:admin;comment:来源(builtin/admin)" json:"source"`
	MarketShared int8   `gorm:"column:market_shared;type:tinyint;not null;default:0;comment:是否共享至市场" json:"marketShared"`
	Deleted      int64  `gorm:"column:deleted;type:bigint;not null;default:0;comment:逻辑删除标识" json:"deleted"`
}

func (SysAiSkill) TableName() string {
	return "sys_ai_skill"
}

// SysAiSkillFile SKILL 目录文件清单（内容存对象存储）
type SysAiSkillFile struct {
	BaseModel
	SkillID  int64   `gorm:"column:skill_id;type:bigint;not null;comment:所属Skill主键" json:"skillId"`
	Path     string  `gorm:"column:path;type:varchar(500);not null;comment:相对SKILL根目录路径" json:"path"`
	FileSize int64   `gorm:"column:file_size;type:bigint;not null;default:0;comment:文件大小(字节)" json:"fileSize"`
	FileType *string `gorm:"column:file_type;type:varchar(64);comment:文件类型" json:"fileType"`
}

func (SysAiSkillFile) TableName() string {
	return "sys_ai_skill_file"
}
